extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -360,6 +360,17 @@ public void handle(Void e) {
                 }
             });
         }
+        // Headers sent on any request, regardless of the response
+        Map<String, String> headers = httpConfiguration.header;
+        if (!headers.isEmpty()) {
+            httpRouteRouter.route().order(Integer.MIN_VALUE).handler(new Handler<RoutingContext>() {
+                @Override
+                public void handle(RoutingContext event) {
+                    event.response().headers().addAll(headers);
```

Comment:
Technically the application could clear this map, which is not uncommon in the case of errors (e.g. if you are sending an error response you might clear this to make sure nothing is left over from the failed response).

I don't know how much this matters.

Reply:
Well, the primary goal of this feature is to set HTTP Headers in the response without having to write additional code. I think in the use case you described this would be still valid.

Reply:
At least, the application can check the headers and decide if they need to be removed or not (even is most of the time, we just call `clear`).